packages/dynamite/dynamite/lib/src/models/openapi/parameter.dart

```diff
@@ -54,9 +59,111 @@ abstract class Parameter implements Built<Parameter, ParameterBuilder> {
     return null;
   }
 
+  /// Builds the uri template value for this parameter.
+  ///
+  /// Throws a StateError if the parameter does not support a uri template.
+  String uriTemplate({
+    final bool isFirst = false,
+    final bool withPrefix = true,
+  }) {
+    final error = 'The parameter $name is not supported by RFC 6570 UriTemplates. Dynamite can not yet work with it.';
+    final buffer = StringBuffer();
+
+    final prefix = switch (style) {
+      ParameterStyle.simple => null,
+      ParameterStyle.label => '.',
+      ParameterStyle.matrix => ';',
+      ParameterStyle.form => isFirst ? '?' : '&',
+      ParameterStyle.spaceDelimited || ParameterStyle.pipeDelimited => explode
+          ? isFirst
+              ? '?'
+              : '&'
+          : throw UnsupportedError(error),
+      ParameterStyle.deepObject || _ => throw UnsupportedError(error),
+    };
+
+    if (prefix != null && withPrefix) {
+      buffer.write(prefix);
+    }
+
+    if (allowReserved) {
+      buffer.write('+');
+    }
+
+    buffer.write(pctName);
+
+    if (explode) {
+      buffer.write('*');
+    }
+
+    return buffer.toString();
+  }
+
+  /// The pct encoded name of this parameter.
+  String get pctName => Uri.encodeQueryComponent(name);
```

Comment:
```suggestion
  String get percentEncodedName => Uri.encodeQueryComponent(name);
```
It's better to use full names instead of abbreviations

Reply:
I disagree. The RFC6570 only calls them `pct-encoded`

Reply:
Then `pctEncodedName` please